app/src/main/java/dev/msfjarvis/lobsters/data/repo/LobstersRepository.kt

```diff
@@ -0,0 +1,37 @@
+package dev.msfjarvis.lobsters.data.repo
+
+import dev.msfjarvis.lobsters.data.local.LobstersPost
+import dev.msfjarvis.lobsters.database.LobstersDatabase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class LobstersRepository @Inject constructor(private val lobstersDatabase: LobstersDatabase) {
+
+  fun isPostSaved(postId: String): Boolean {
+    val post = lobstersDatabase.postQueries.isPostSaved(postId).executeAsOneOrNull()
+    return post != null
+  }
```

Comment:
why are the others suspending but not this?

Reply:
Fixed